app/controllers/ViewReturnSummaryController.scala

```diff
@@ -50,7 +50,9 @@ class ViewReturnSummaryController @Inject() (
           throw new IllegalStateException("no enrolmentId, all users at this point should have one")
         )
 
-        val submittedReturnF: Future[ReturnDisplayApi] = taxReturnHelper.fetchTaxReturn(pptId, periodKey.toUpperCase())
+        if (!periodKey.matches("[A-Z0-9]{4}")) throw new Exception(s"Period key '$periodKey' is not allowed.")
```

Comment:
Do we still need t store the period key separately now we are storing the tax return API in its entirety? Can we just always use the one in the charge details object? 

![Screenshot 2022-05-03 at 10 44 36](https://user-images.githubusercontent.com/40767309/166433531-e5cffb60-d8ce-4e5a-b33b-d6e987570e56.png)

Reply:
probably not, I could do this refactor in this pr? However this ticket is for ensuring the user hasnt tampered with the url  which is where this period key is coming from